RockWeb/Themes/NewSpring/assets/Lava/DATASETS/content-channel-items-by-channel.lava

```diff
@@ -54,95 +54,18 @@
           AND av.AttributeId = 86072
           AND av2.AttributeId IN (86070,86071)
         {%- endsql -%}
+
         {%- if childchannelid and childchannelid != empty -%}
-          {%- sql return:'children' -%}
-            SELECT
-              ROW_NUMBER() OVER (ORDER BY ccia.[Order]) 'Index',
-              cci.Id,
-              cci.ContentChannelId 'ChannelId',
-              cci.StartDateTime 'PublishDateTime',
-              cci.ExpireDateTime 'ExpireDateTime',
-              ccis.Slug,
-              pd.AccessKey 'Dataset'
-            FROM ContentChannelItemAssociation ccia
-            JOIN ContentChannelItem cci
-            ON ccia.ChildContentChannelItemId = cci.Id
-            JOIN ContentChannelItemSlug ccis
-            ON ccis.ContentChannelItemId = cci.Id
-            JOIN AttributeValue av
-            ON av.EntityId = cci.ContentChannelId
-            JOIN PersistedDataset pd
-            ON pd.Id = av.Value
-            WHERE ccia.ContentChannelItemId = {{ item.Id }}
-            AND cci.ContentChannelId = {{ childchannelid }}
-            AND av.AttributeId = 107096
-            {% if childItemsManuallyOrdered == 'true' %}
-              ORDER BY ccia.[Order]
-            {% else %}
-              ORDER BY cci.StartDateTime
-            {% endif %}
-          {%- endsql -%}
+          {% assign childchannelid = childchannelid | AsInteger %}
+          {% assign children = 'content_children' | PersistedDataset | Where:'Id', item.Id | Select:'Children' | First | Where:'ChannelId', childchannelid %}
         {%- endif -%}
+
         {%- if parentchannelid and parentchannelid != empty -%}
-          {%- sql return:'parents' -%}
-            SELECT
-              ROW_NUMBER() OVER (ORDER BY ccia.[Order]) 'Index',
-              cci.Id,
-              cci.ContentChannelId 'ChannelId',
-              cci.StartDateTime 'PublishDateTime',
-              cci.ExpireDateTime 'ExpireDateTime',
-              ccis.Slug,
-              pd.AccessKey 'Dataset'
-            FROM ContentChannelItemAssociation ccia
-            JOIN ContentChannelItem cci
-            ON ccia.ContentChannelItemId = cci.Id
-            JOIN ContentChannelItemSlug ccis
-            ON ccis.ContentChannelItemId = cci.Id
-            JOIN AttributeValue av
-            ON av.EntityId = cci.ContentChannelId
-            JOIN PersistedDataset pd
-            ON pd.Id = av.Value
-            WHERE ccia.ChildContentChannelItemId = {{ item.Id }}
-            AND cci.ContentChannelId = {{ parentchannelid }}
-            AND av.AttributeId = 107096
-            ORDER BY ccia.[Order]
-          {%- endsql -%}
+          {% assign parentchannelid = parentchannelid | AsInteger %}
+          {% assign parents = 'content_parents' | PersistedDataset | Where:'Id', item.Id | Select:'Parents' | First | Where:'ChannelId', parentchannelid %}
```

Comment:
Getting parent data from the parents dataset vs making another sql call. This dramatically speeds up build time.